File: Python Crawler/source.py
class Source:
    def __init__(self):
        self._dataFormat = ""
        self._name = ""
        self._type = ""
        self._url = ""

    def __init__(self, df, name, sourcetype, url):
        self._dataFormat = df
        self._name = name
        self._type = sourcetype
        self._url = url

    def get_name(self):
        return self._name

    def get_data_format(self):
        return self._dataFormat

    def get_type(self):
        return self._type

    def get_url(self):
        return self._url

    def set_name(self, name):
        self._name=name

    def set_data_format(self, df):
        self._dataFormat=df

    def set_url(self, url):
        self._url=url

    def set_type(self, sourcetype):
        self._type=sourcetype

File: SpringBoot App/heterogeneity/src/main/java/com/threatgator/heterogeneity/model/Source.java
package com.threatgator.heterogeneity.model;
import java.util.List;

// Source Model Class

public class Source {
    @Override
    public String toString() {
        return "Source{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", dataFormat='" + dataFormat + '\'' +
                ", sourceType='" + sourceType + '\'' +
                ", url='" + url + '\'' +
                ", accountsList=" + accountsList +
                '}';
    }

    protected int id; // Primary key
    // Other table entries
    protected String name;
    protected String dataFormat;
    protected String sourceType;
    protected String url;
    // defined relationship

//    @JoinColumn(name = "src_ac_fk",referencedColumnName = "id")
    private List<Account> accountsList;

    public Source() {

    }
    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDataFormat() {
        return dataFormat;
    }

    public void setDataFormat(String dataFormat) {
        this.dataFormat = dataFormat;
    }

    public String getSourceType() {
        return sourceType;
    }

    public void setSourceType(String sourceType) {
        this.sourceType = sourceType;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }
}
